src/config.h

```diff
@@ -157,3 +157,33 @@
 #ifndef MG_ENABLE_PROFILE
 #define MG_ENABLE_PROFILE 0
 #endif
+
+#ifndef MG_ENABLE_DRIVER_INIT
+#define MG_ENABLE_DRIVER_INIT 1
+#endif
+
+#ifndef MG_TCPIP_IP
+#define MG_TCPIP_IP mg_htonl(MG_U32(0, 0, 0, 0))
+#endif
+
+#ifndef MG_TCPIP_MASK
+#define MG_TCPIP_MASK mg_htonl(MG_U32(0, 0, 0, 0))
+#endif
+
+#ifndef MG_TCPIP_GW
+#define MG_TCPIP_GW mg_htonl(MG_U32(0, 0, 0, 0))
+#endif
+
+#define MG_MAC_ADDRESS_RANDOM \
+  { 0, 0, 0, 0, 0, 0 }
+
+#define MG_MAC_ADDRESS_STM32_UID(ADDR)                       \
+  {                                                          \
+    2, ((uint8_t *) (ADDR))[0] ^ ((uint8_t *) (ADDR))[1],    \
+        ((uint8_t *) (ADDR))[2] ^ ((uint8_t *) (ADDR))[3],   \
+        ((uint8_t *) (ADDR))[4] ^ ((uint8_t *) (ADDR))[5],   \
+        ((uint8_t *) (ADDR))[6] ^ ((uint8_t *) (ADDR))[7] ^  \
+            ((uint8_t *) (ADDR))[8],                         \
+        ((uint8_t *) (ADDR))[9] ^ ((uint8_t *) (ADDR))[10] ^ \
+            ((uint8_t *) (ADDR))[11]                         \
+  }\
```

Comment:
Some STM32s do not allow byte reading, some Hs require word reading.
This actually depends on the processor. We can probably add this to the driver header files and deal with processor differences there.
I don't see this belonging to config.h